controllers/hvpa_controller_test.go

```diff
@@ -268,293 +285,342 @@ var _ = Describe("#TestReconcile", func() {
-							{
-								Type:   autoscaling.ScalingLimited,
-								Status: v1.ConditionTrue,
-							},
-						}),
-					vpaStatus: newVpaStatus("deployment", "3G", "200m"),
-					target:    target,
-					vpaWeight: autoscalingv1alpha1.VpaWeight(40),
+					hvpa:      newHvpa("hvpa-2", target.GetName(), "label-2", minChange),
+					hpaStatus: nil,
+					vpaStatus: newVpaStatus("deployment", "0.8G", "150m"),
+					target:    newTarget("deployment", unscaledLarge, 3),
 				},
 				action: action{
-					maintenanceWindow: &autoscalingv1alpha1.MaintenanceTimeWindow{
-						Begin: utils.NewMaintenanceTime(time.Now().UTC().Hour()-1, 0, 0).Formatted(),
-						End:   utils.NewMaintenanceTime(time.Now().UTC().Hour()+1, 0, 0).Formatted(),
+					maintenanceWindow: &hvpav1alpha1.MaintenanceTimeWindow{
+						Begin: utils.NewMaintenanceTime((time.Now().UTC().Hour()-1)%24, 0, 0).Formatted(),
+						End:   utils.NewMaintenanceTime((time.Now().UTC().Hour()+1)%24, 0, 0).Formatted(),
 					},
-					updateMode:   autoscalingv1alpha1.UpdateModeMaintenanceWindow,
+					updateMode:   hvpav1alpha1.UpdateModeMaintenanceWindow,
 					limitScaling: limitScale,
 				},
 				expect: expect{
-					scalingOff:      false,
-					desiredReplicas: 3,
+					desiredReplicas: 2,
 					resourceChange:  true,
-					scaleOutLimited: true,
-					resources:       scaledByWeight40,
-					blockedReasons: []autoscalingv1alpha1.BlockingReason{
-						autoscalingv1alpha1.BlockingReasonMinChange,
-					},
+					resources:       scaledLarge,
+					blockedReasons:  []hvpav1alpha1.BlockingReason{},
 				},
 			}),
-			Entry("UpdateMode Auto, HPA scaling not limited", &data{
+			Entry("VPA unsupported condition", &data{
 				setup: setup{
-					hvpa: newHvpa("hvpa-2", target.GetName(), "label-2", minChange),
-					hpaStatus: newHpaStatus(
-						2, 3, []autoscaling.HorizontalPodAutoscalerCondition{
-							{
-								Type:   autoscaling.ScalingLimited,
-								Status: v1.ConditionFalse,
-							},
-						}),
+					hvpa:      newHvpa("hvpa-2", target.GetName(), "label-2", minChange),
+					hpaStatus: nil,
 					vpaStatus: newVpaStatus("deployment", "3G", "500m"),
 					target:    target,
-					vpaWeight: autoscalingv1alpha1.VpaWeight(40),
 				},
-				expect: expect{
-					scalingOff:      false,
-					desiredReplicas: 3,
-					resourceChange:  false,
-					scaleOutLimited: false,
-					resources:       unscaled,
-					blockedReasons: []autoscalingv1alpha1.BlockingReason{
-						autoscalingv1alpha1.BlockingReasonUpdatePolicy,
+				action: action{
+					vpaStatusCondition: []vpa_api.VerticalPodAutoscalerCondition{
+						{
+							Type:   vpa_api.ConfigUnsupported,
+							Status: v1.ConditionTrue,
+						},
 					},
 				},
+				expect: expect{
+					resourceChange: false,
+					blockedReasons: []hvpav1alpha1.BlockingReason{},
+				},
 			}),
-			Entry("UpdateMode Auto, Proportional scaling", &data{
+			Entry("UpdateMode Auto, scale down hysteresis", &data{
```

Comment:
```suggestion
			Entry("UpdateMode Auto, scale down hysteresis based on scaling intervals overlap", &data{
```

Reply:
Is the scale up side of this covered?

Reply:
Can we add a couple of cases (scale up and down) without scaling interval gaps configured?

Reply:
scale down hysteresis due to `ScalingIntervalsOverlap` is covered in this case. scale up hysteresis is provided by `minChange` which is covered in test case - `UpdateMode Auto, scale up blocked due to minChange`

Reply:
> scale up hysteresis is provided by minChange which is covered in test case - UpdateMode Auto, scale up blocked due to minChange

Understood. But I meant a specific case for scale up hysteresis where minChange doesn't block. I.e. scale up happens within the current bucket even though the next bucket's min total resources is higher than the scaled up total resources. WDYT?

Reply:
ah ok.. sure. will add.